src/encode/pattern/mod.rs

```diff
@@ -688,6 +699,21 @@ impl Deserialize for PatternEncoderDeserializer {
     }
 }
 
+/// Get the process ID as String for different OS
+#[cfg(not (target_os = "windows"))]
+fn get_pid() -> String { 
```

Comment:
I don't think process IDs can ever be negative. In any case, i64 would also work fine.